parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -793,6 +793,18 @@ configGPUDrivers() {
         echo "os $OS not supported at this time. skipping configGPUDrivers"
         exit 1
     fi
+    
+    if [[ "$NVIDIA_DRIVER_IMAGE" == *"grid"* ]]; then
+        if [ -x /usr/bin/nvidia-gridd ]; then
+            if nvidia-smi -q | grep -q "License Status.*Unlicensed"; then
+                echo "GRID license status is Unlicensed. Restarting nvidia-gridd..."
+                sudo pkill nvidia-gridd || true
+                sudo /usr/bin/nvidia-gridd &
```

Comment:
I'm surprised nvidia-gridd doesn't run as a system service. Should it be? That way it'll restart automatically.

Reply:
Yes, I was thinking about that too. In fact, this code assumes there's a systemd unit: https://github.com/Azure/aks-gpu/blob/0adfc0c125107c12618153d916015326b7ef1247/install.sh#L77, but when i checked logs it doesn't have a systemd unit defined.

Reply:
I also think it's worth defining it as a systemd unit in the future. This is a forced restart post kernel module reload. So we might need an explicit force restart even then.